app.rb

```diff
@@ -0,0 +1,153 @@
+    when 4 then create_book
+    when 5 then create_rental
+    when 6 then list_rentals
+    when 7 then exit(true)
+    else
+      puts 'PLEASE ENTER A NUMBER (1..7)'
+    end
+  end
+  # rubocop:enable Style/CyclomaticComplexity
+
+  def list_books
+    @books.each do |book|
+      puts "Title: #{book.title}, Author: #{book.author}"
+    end
+  end
+
+  def list_people
+    @person.each do |p|
+      puts "[#{p.class.name}] Name: #{p.name}, ID: #{p.id}, Age: #{p.age}"
+    end
+  end
+
+  def create_book
+    print 'Title:'
+    title = gets.chomp
+    print 'author:'
+    author = gets.chomp
+    @books.push(Book.new(title, author))
+    puts 'Book Created Successfully'
+  end
+
+  def check_number(msg)
+    number = 0
+    loop do
+      print msg
+      input = gets.chomp.to_i
+      if input.is_a?(Integer) && input.positive?
+        number = input
+        break
+      else
+        puts 'Please, enter a valid input!'
+      end
+    end
+    number
+  end
+
+  def check_options(msg, options)
+    number = 0
+    loop do
+      print msg
+      input = gets.chomp.to_i
+      if options.include?(input)
+        number = input
+        break
+      else
+        puts 'Please, enter a valid input!'
+      end
+    end
+    number
+  end
+
+  def check_permission(permission)
+    case permission
+    when 'y' then permission = true
+    when 'n' then permission = false
+    end
+    permission
+  end
+
+  def create_person
+    num = check_options('Do you want to create a student (1) or a teacher (2)? [input the number]: ', [1, 2])
+    age = check_number('Age:')
+
+    print 'name:'
+    name = gets.chomp
+    classroom = check_number('Classroom:')
+
+    case num
+    when 1
+      print 'Has parent permission? [y/n]:'
+      permission = gets.chomp
+
+      permission = check_permission(permission)
+
+      @person.push(Student.new(age, name, classroom, parent_permission: permission))
+    when 2
+      print 'Specialization:'
+      specialty = gets.chomp
+
+      @person.push(Teacher.new(age, specialty, name: name))
+    else
+      puts 'Invalid number, please enter number again!'
+    end
+    puts 'Person created successfully'
+  end
+
+  def create_rental
+    puts 'Select a book from the following list'
+    @books.each_with_index do |book, index|
+      puts "#{index + 1}) Title: #{book.title}, Author: #{book.author}"
+    end
+    book_num = gets.chomp.to_i
+
+    puts 'Select a person from the following list'
+    @person.each_with_index do |per, index|
+      puts "No: #{index + 1}, [#{per.class}] Name: #{per.name}, ID: #{per.id}, Age: #{per.age}"
+    end
+    puts @person[0].name
+    iam = gets.chomp.to_i
+    puts iam
+    puts "HERE #{@person[iam - 1].name}"
+    puts @person[iam - 1].name
+    print 'Date:'
+    date = gets.chomp
+
+    p_index = iam - 1
+    p @person[p_index]
+    @rentals.push(Rental.new(date, @books[book_num - 1], @person[p_index]))
+    puts 'Rental Created successfully'
```

Comment:
When trying to create a rental, your app displays the whole object and other information different from that of the example video as shown below.  Kindly refactor your code to show the same UI or similar. :wink: 

### Your UI

![Screenshot from 2023-02-02 11-09-15](https://user-images.githubusercontent.com/65233860/216310385-9f3094b4-9947-412a-aece-ab6676bda454.png)



### Actual display 

![Screenshot from 2023-02-02 11-09-36](https://user-images.githubusercontent.com/65233860/216310312-3d5e992e-33cb-4d7b-b47c-d9982c5de3c2.png)

Reply:
resolved! ✔️